src/middlewared/middlewared/plugins/ups.py

```diff
@@ -168,7 +169,7 @@ async def validate_data(self, data, schema):
 
         driver = data.get('driver')
         if driver:
-            if driver not in (await self.middleware.call('ups.driver_choices')).keys():
+            if driver not in (await self.middleware.call('ups.driver_choices')).keys() and driver != NUT_DUMMY_UPS:
```

Comment:
Would it be possible to instead just mock over `ups.driver_choices` in our jenkins CI runs?

Reply:
This should probably do the job https://github.com/truenas/middleware/blob/bf02e80889e1d491023c6e9de58643c149f849e8/src/middlewared/middlewared/test/integration/utils/mock.py#L11